### DISCUSSION TITLE
Vorstellung xCalenadar Projekt

### DISCUSSION THREAD

**Discussion by PetrusQn** [MelonigeMelone/xCalendar#2] [Blogkanal]
Viele Projekte scheitern aktuell an mangelhafter Kommunikation innerhalb des Projektteams. Häufig ist der Grund dafür das Wechseln zwischen verschiedenen Anwendungen im Arbeitsalltag. So werden Nachrichten, Arbeitsaufträge oder Aktualisierungen von Forderungen schnell übersehen. 
Im Rahmen des "xCalendar"-Projekts soll diesem Problem, auf privater und geschäftlicher Ebene, entgegengewirkt werden. xCalendar ist eine Kalender-Anwendung mit einigen zusätzlichen Kniffen um die Terminplanung und Organisation zu vereinfachen. Dazu zählen Funktionen wie das Synchronisieren bzw. Teilen des eigenen Kalenders oder expliziten Aktivitäten mit anderen Personen. Des Weiteren gibt es Aktivitätsbezogene Chatgruppen, sodass keine Drittanbieter Nachrichten-App verwendet werden muss. Datensch(m)utz ist kein Thema :-) Neben den Standard-Funktionen die jede Kalender-App bietet, bietet xCalendar die Möglichkeit einen Ort, pro Aktivität, zu hinterlegen. Der Benutzer kann sich anschließend über eine Anbindung an Google- oder Apple Maps (noch nicht beschlossen) zu dem hinterlegten Ort navigieren lassen. Damit der Nutzer sich für ein angebrachtes Fortbewegungsmittel entscheiden kann und nicht im Regen laufen muss, gibt es zusätzlich eine Wetter-Funktion. 

Zur Umsetzung des Projekts werden folgende Technologien und Plattformen verwendet:

- Jira 
- Jenkins
- GitHub
- Java 17 
- Angular
- SQL (Datenbankangaben folgen)
- GraphQL

Das Team besteht aus David Abert (Entwickler), David Bohr (Entwickler und Product Owner), Tobias Eberle (Backend-Entwickler und Scrum-Master), Paul Kuhn (Entwickler). Wir treffen uns alle zwei bis drei Tage um den aktuellen Stand und Probleme zu besprechen. Nach zwei Wochen treffen wir uns um den vergangenen Sprint zu reflektieren und den nächsten zu planen. 

Projektregeln:
Sprintlänge: 2 Wochen (neuer Sprint startet immer Montags)
SprintChangeMeeting: am Wochenende vor dem Sprintwechsel (Review -> Retro -> Sprintplanung) 
Daily: Dienstag und Freitag 15Uhr (15-20min)

Der Backlog des ersten Sprints setzt sich wie folgt zusammen:

![image](https://github.com/MelonigeMelone/xCalendar/assets/136441045/5a0d603b-69a6-40e1-8f9d-88c0f5f62a84)

Aktuelles Kanban-Board:
  
![image](https://github.com/MelonigeMelone/xCalendar/assets/136441045/671dd2e9-956f-429c-a84b-dce0a8691264)

**Comment by Christian-2003** [comment]
Hallo zusammen,

da habt ihr einen wirklich schönen, kurzen und knackigen Blogeintrag geschrieben. Weiter so 👍🏼. Im Grunde genommen, ist ja alles enthalten, was gewünscht war.

Besonders schön finde ich eure Einleitung. Dass man wichtige Nachrichten verpasst, kann wahrscheinlich jeder nachvollziehen. Da könnte euer Kalender definitiv eine wichtige Marktlücke schließen.
Da ihr ja schon auf GitHub seid, wäre es schön, wenn ihr vielleicht etwas mehr Markdown bei solchen Blogeinträgen verwendet, dann sieht sowas (visuell) etwas ansprechender aus.

Was mir auch gefällt ist, dass ihr sowohl bereits eure Rollen eingeteilt, als auch euer Jira-Projekt aufgesetzt habt. Wenn ich mir eure Tasks für den ersten Sprint anschaue, dann erkenne ich, dass ihr euch so einiges vorgenommen habt für den ersten Sprint. Respekt.
In einem der Tasks steht, dass ihr ein IOS-Projekt aufsetzen wollt. Habt ihr also vor den Kalender für IOS zu entwickeln, oder programmiert ihr zusätzlich noch bspw. eine Webseite?

Als letztes hätte ich noch eine reine Interessensfrage: Wie lange dauern eure Sprints? Das würde mich interessieren, damit ich die Entscheidung unserer Projektgruppe diesbezüglich besser einordnen kann.

Insgesamt habt ihr einen wirklich interessanten Blogeintrag geschrieben. Ich freue mich bereits darauf, euren nächsten Eintrag zu lesen.

**Reply by MelonigeMelone** [reply]
Vielen Dank für euer Feedback. 

Um kurze auf eure Fragen einzugehen, ja wir wollen einerseits eine Website sowie eine IOS App machen. Die Website wollen wir mit Angular realisieren. Da wir uns bei der IOS App noch nicht auf den TechStack 100% geeinigt haben, gilt es dort erst noch ein wenig Forschungsarbeit zu betreiben, um die optimale Technologie zu wählen.

Auf deine zweite Frage unsere Sprints haben eine Dauer von 2 Wochen.

**Comment by Joh4nta** [comment]
Hallo ihr,
wir hätten die Frage, wieso ihr im Sprint keine Tasks, sondern nur User-Storys verwendet. Außerdem fehlen uns die Zuordnung von den einzelnen Personen für die User-Storys. Ist der Orga-Kram wo anders gespeichert? Wäre schön, wenn ihr das hier auch posten würdet. 
Ansonsten gefällt uns eure Idee und wir sind gespannt.

**Reply by MelonigeMelone** [reply]
Vielen Dank für euer Feedback.

Um die erste Frage zu beantworten, dass hatten wir wohl vergessen zu Ändern. Beim Erstellen wurde automatisch der Typ User-Story ausgewählt, aber ja eigentlich sind es Tasks. Dies werden wir noch ändern, danke für den Hinweis.

Und auf die zweite Frage, wir haben gezielt noch keine Personen an die Tasks zugeordnet, welche alle im Team übernehmen können. Lediglich die Task bei welchem wir gerne eine bestimmte Person zugeordnet haben wollten, da diese sich besonders gut in diesem Gebiet auskennt, haben wir auch Personen zugeordnet. 
Die restlichen Aufgaben werden unter den anderen Personen aufgeteilt, je nachdem wie viel Zeit die Personen haben. Dazu klären wir regelmäßig in den Dailys den aktuellen Stand und die nächsten Aufgaben die anstehen.

Eine kurze Übersicht über Sprintlänge etc. werden wir noch hinzufügen, wenn das mit Organisation gemeint ist.
